graph/network/diffusion_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package network
+
+import (
+	"math"
+	"testing"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+var heatDiffusionTests = []struct {
```

Comment:
I think it would be worth adding two test cases:
* Initial "temperature" `h` is constant. `HeatDiffusion` then should not change `h`.
* Initial "temperature" `h` is not constant but the average temperate should be constant independently of `t`.

Reply:
Done. They are not happy tests due to the instability of `mat.Exp`. Worse for the symmetric normalised Laplacian (so I avoid that completely for the global temperature).